Jenkinsfile-sgx-hw

```diff
@@ -0,0 +1,107 @@
+#!groovy
+
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------------
+
+pipeline {
+    agent {
+        node {
+            label 'slave-1'
+            customWorkspace "workspace/${env.BUILD_TAG}"
+        }
+    }
+
+    options {
+        timestamps()
+        buildDiscarder(logRotator(daysToKeepStr: '31'))
+    }
+
+    environment {
+        ISOLATION_ID = sh(returnStdout: true,
+                          script: 'printf $BUILD_TAG | sha256sum | cut -c1-64').trim()
+        COMPOSE_PROJECT_NAME = sh(returnStdout: true,
+                          script: 'printf $BUILD_TAG | sha256sum | cut -c1-64').trim()
+    }
+
+    stages {
+        stage('Check for Signed-Off Commits') {
+            steps {
+                sh '''#!/bin/bash -l
+                    if [ -v CHANGE_URL ] ;
+                    then
+                        temp_url="$(echo $CHANGE_URL |sed s#github.com/#api.github.com/repos/#)/commits"
+                        pull_url="$(echo $temp_url |sed s#pull#pulls#)"
+
+                        IFS=$'\n'
+                        for m in $(curl -s "$pull_url" | grep "message") ; do
+                            if echo "$m" | grep -qi signed-off-by:
+                            then
+                              continue
+                            else
+                              echo "FAIL: Missing Signed-Off Field"
+                              echo "$m"
+                              exit 1
+                            fi
+                        done
+                        unset IFS;
+                    fi
+                '''
+            }
+        }
+
+        stage('Run Lint') {
+            steps {
+                sh 'docker-compose -f docker-compose-lint.yaml up --exit-code-from avalon-lint'
+                sh './../sgx-hw.sh config/singleton_enclave_config.toml'
```

Comment:
Is `sgx-hw.sh` a new file.  It's missing in this PR

Reply:
this file is placed in slave machine. It is used to handle usage of specific spid and ias_api_key parameters in enclave config file

Reply:
But still it make sense to include in Git repository since without this SGX HW CI build will not be functional

Reply:
No need. This script is placed in slave machine. Jenkins will trigger the SGX-HW job on the slave where this script is already present. I have tested it and it is working.

Reply:
I mean we don't have a separate repository to maintain the scripts needed by Jenkins slave machines. So we can add all Jenkins server and slave related files in this repo. We can add a README which can mention about slave, server files.